web/tests/stream_list_sort.test.cjs

```diff
@@ -217,6 +283,128 @@ test("basics", () => {
+    assert.deepEqual(sorted_sections[2].section_title, "FRONTEND");
+    assert.deepEqual(sorted_sections[3].id, "normal-streams");
+    assert.deepEqual(sorted_sections[3].section_title, "translated: OTHER");
+
+    // If both `pin_to_top` is true and folder_id is set, as in
+    // the channel `scalene`, then the channel ends up in the pinned
+    // section and `folder_id` is ignored.
+    assert.deepEqual(sorted_sections[0].streams, [scalene.stream_id]);
+    assert.deepEqual(sorted_sections[0].muted_streams, [muted_pinned.stream_id]);
+    assert.deepEqual(sorted_sections[0].inactive_streams, []);
+    assert.deepEqual(sorted_sections[1].streams, [stream_hyphen_underscore_slash_colon.stream_id]);
+    assert.deepEqual(sorted_sections[1].muted_streams, []);
+    assert.deepEqual(sorted_sections[1].inactive_streams, []);
+    assert.deepEqual(sorted_sections[2].streams, [fast_tortoise.stream_id]);
+    assert.deepEqual(sorted_sections[2].muted_streams, [muted_active.stream_id]);
+    assert.deepEqual(sorted_sections[2].inactive_streams, [pneumonia.stream_id]);
+    assert.deepEqual(sorted_sections[3].streams, [clarinet.stream_id]);
+    assert.deepEqual(sorted_sections[3].muted_streams, []);
+    assert.deepEqual(sorted_sections[3].inactive_streams, []);
+
+    // The first and last sections are invariant. The intermediate sections
+    // are arranged by the `order` field in the channel folder object.
+    assert.deepEqual(stream_list_sort.section_ids(), [
+        "pinned-streams",
+        "2",
+        "1",
+        "normal-streams",
+    ]);
+});
+
+test("current_section_id_for_stream", ({override}) => {
+    add_all_subs();
+
+    sort_groups("");
+    assert.equal(
+        stream_list_sort.current_section_id_for_stream(scalene.stream_id),
+        "pinned-streams",
+    );
+    assert.equal(
+        stream_list_sort.current_section_id_for_stream(clarinet.stream_id),
+        "normal-streams",
+    );
+    assert.equal(
+        stream_list_sort.current_section_id_for_stream(fast_tortoise.stream_id),
+        "normal-streams",
+    );
+    assert.equal(
+        stream_list_sort.current_section_id_for_stream(
+            stream_hyphen_underscore_slash_colon.stream_id,
+        ),
+        "normal-streams",
+    );
+
+    override(user_settings, "web_left_sidebar_show_channel_folders", true);
+    sort_groups("");
+    // Now fast_tortoise should appear in its respective folder.
+    assert.equal(
+        stream_list_sort.current_section_id_for_stream(scalene.stream_id),
+        "pinned-streams",
+    );
+    assert.equal(
+        stream_list_sort.current_section_id_for_stream(clarinet.stream_id),
+        "normal-streams",
+    );
+    assert.equal(stream_list_sort.current_section_id_for_stream(fast_tortoise.stream_id), "1");
+    assert.equal(
+        stream_list_sort.current_section_id_for_stream(
+            stream_hyphen_underscore_slash_colon.stream_id,
+        ),
+        "2",
```

Comment:
It would be nice if instead of numbers here (and a fe other places above) you could instead reference an attribute on a folder object. So when adding folders above with `channel_folders.initialize({` make a `backend_folder` object first, and use that here to give more meaning to this value instead of just`"2"`